airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/DefaultJdbcDatabase.java

```diff
@@ -116,7 +113,12 @@ public DatabaseMetaData getMetaData() throws SQLException {
 
   @Override
   public void close() throws Exception {
-    connectionSupplier.close();
+    if (dataSource instanceof AutoCloseable autoCloseable) {
```

Comment:
tiny nitpick: it would be nice to keep this comment https://github.com/airbytehq/airbyte/pull/10342/files#diff-c7c618583584a27fb3b21cef812ca82b6d077d8bc779cf9a401bfc7658bb8dbaL145-L146

also I didn't know this `instanceof T var` syntax existed! it's pretty cool 🚛

Reply:
Good point. Will add it in a follow-up PR.

Reply:
Done in https://github.com/airbytehq/airbyte/pull/10444.